File: sol/1170/1170.py
from bisect import bisect

class Solution:
    @staticmethod
    def f(s: str) -> int:
        c = '~'
        f = 0
        for x in s:
            if x < c:
                c = x
                f = 1
            elif x == c:
                f += 1
        return f
    def numSmallerByFrequency(self, queries: List[str], words: List[str]) -> List[int]:
        n = len(words)
        refFreq = sorted(self.f(w) for w in words)
        return [
            n - bisect(refFreq, self.f(q))
            for q in queries
        ]

File: sol/1170/1170.cpp
struct Solution {
    inline static int f(string &s) {
        char c = '~';
        int freq = 0;
        for (char x : s){
            if (x < c){
                c = x;
                freq = 1;
            } else if (x == c){
                ++freq;
            }
        }
        return freq;
    }
    vector<int> numSmallerByFrequency(vector<string>& queries, vector<string>& words) {
        int n = words.size();
        vector<int> refFreq(n), res(queries.size());
        ranges::transform(words, refFreq.begin(), this->f);
        ranges::sort(refFreq);
        ranges::transform(queries, res.begin(), [&](string &q){
            return n - ranges::distance(refFreq.begin(), ranges::upper_bound(refFreq, this->f(q)));
        });
        return res;
    }
};
